src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -193,12 +193,20 @@ public void init(AdaptorContext context) throws Exception {
       log.config("adaptor runs in lister-only mode");
     }
 
-    modeOfOperation = cfg.getValue("db.modeOfOperation");
+    modeOfOperation = cfg.getValue("db.modeOfOperation").trim();
     if ("urlAndMetadataLister".equals(modeOfOperation) && encodeDocId) {
       String errmsg = "db.modeOfOperation of \"" + modeOfOperation
           + "\" requires docId.isUrl to be \"true\"";
       throw new InvalidConfigurationException(errmsg);
     }
+    String modeOfOperationValid[] = {"blobColumn", "filepathColumn",
```

Comment:
Thanks for your clarification. I changed the error message in loadResponseGenerator using the reflection mechanism to also return the valid methods names.